selfdrive/car/volkswagen/carcontroller.py

```diff
@@ -65,6 +65,15 @@ def update(self, CC, CS, ext_bus, now_nanos):
       self.apply_steer_last = apply_steer
       can_sends.append(self.CCS.create_steering_control(self.packer_pt, CANBUS.pt, apply_steer, hca_enabled))
 
+      if self.CP.flags & VolkswagenFlags.STOCK_HCA_PRESENT:
+        # Pacify VW Emergency Assist driver inactivity detection by changing its view of driver steering input torque
+        # to the greatest of actual driver input or 2x openpilot's output (1x openpilot output is not enough to
+        # consistently reset inactivity detection on straight level roads). See commaai/openpilot#23274 for background.
```

Comment:
can we also consider controlsd's force_decel?

Reply:
In stock ACC mode, with a gateway harness, unfortunately we're in the wrong bus position to MITM the command that changes the ACC setpoint to zero. Otherwise we'd mitigate the problem that way instead.

In openpilot longitudinal mode it'll be very possible to reimplement the VW Emergency Assist experience keyed off force_decel (among other inputs) but that's an entirely different PR for another day.

Reply:
Technically we could just NOT run the torque input mitigation if in force_decel, but EA would only kick in 30-60 seconds after we stop mitigating, if it kicks in at all. I think it would be much too unreliable to set any user/code expectation of it working.

Reply:
gonna merge this, but I suspect there should be a way to do it for the stock ACC case. what if we only reset it at the minimum interval conditional on attentiveness?

Reply:
Right now the chance of EA kicking in (before this mitigation) were kinda random, it hits some people more than others. I suppose we could do it keyed on force_decel and then use the mitigation code to send zero driver torque to increase the probability EA is triggered. But it would only work on the subset of cars configured for EA.

I would rather spend the time on a quality EA experience for openpilot longitudinal. Of course it already works to slow the car, but I'd like to add back the UI alerts, the brake-jerk/seatbelt-jerk functions, and the hazard-light function.

Reply:
sounds good